internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/meta/RequestBasedMetadataIntegrationTest.java

```diff
@@ -114,6 +145,32 @@ public void testMetadata() {
     }
   }
 
+  @Test(timeOut = TIME_OUT)
+  public void testMetadataSchemaEvolution() {
+    ReadOnlySchemaRepository schemaRepository = veniceCluster.getRandomVeniceRouter().getSchemaRepository();
+
+    veniceCluster
+        .useControllerClient(controllerClient -> controllerClient.addValueSchema(storeName, EVOLVED_VALUE_SCHEMA));
+
+    Stream<Map.Entry> genericRecordStream = IntStream.range(0, KEY_COUNT).mapToObj(i -> {
+      GenericRecord record = new GenericData.Record(formatSchema(EVOLVED_VALUE_SCHEMA));
+      record.put("id", i);
+      record.put("age", i);
+      return new AbstractMap.SimpleEntry<>(i, record);
+    });
+
+    veniceCluster.createVersion(storeName, KEY_SCHEMA, EVOLVED_VALUE_SCHEMA, genericRecordStream);
+
+    TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, () -> {
```

Comment:
what is the ideal time we are looking at for the metadata to be updated based on the current implementation (periodic refresh)?

Reply:
It's set to 60 seconds normally but in tests I have it set to 1 second. I was actually thinking 30 seconds is probably overkill but would just setting it to 1 second be too little and leave room for flakiness?

Reply:
yes 1 might be flaky, but 30secs is too long. If we set it to a few seconds, eg: 3 and if number are invocations are set to be multiple, we can tell whether the metadata is updated reliably&quickly like it should. You can test locally multiple times to adjust the flakiness of the timeouts and explore further if timeout of few seconds turns out to be flaky when the refresh time is 1 secs which potentially might boil down to actual bugs. IMO, I think we should find a balance between being flaky and being too lenient on the timeout.

Reply:
Got it I'll experiment a bit and also try the same with the other 30 sec assertions in this class